drivers/video/fb.c

```diff
@@ -544,6 +548,79 @@ static int fb_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
   return ret;
 }
 
+/****************************************************************************
+ * Name: fb_poll
+ *
+ * Description:
+ *   Wait for framebuffer to be writable.
+ *
+ ****************************************************************************/
+
+static int fb_poll(FAR struct file *filep, struct pollfd *fds, bool setup)
+{
+  FAR struct inode *inode;
+  FAR struct fb_chardev_s *fb;
+
+  /* Get the framebuffer instance */
+
+  DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+  inode = filep->f_inode;
+  fb    = (FAR struct fb_chardev_s *)inode->i_private;
+
+  if (setup)
+    {
+      if (fb->fds == NULL)
+        {
+          fb->fds = fds;
+          fds->priv = &fb->fds;
+        }
+      else
+        {
+          return -EBUSY;
+        }
+    }
+  else if (fds->priv)
+    {
+      fb->fds = NULL;
+      fds->priv = NULL;
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: fb_pollnotify
+ *
+ * Description:
+ *   Notify the waiting thread that the framebuffer can be written.
+ *
+ * Input Parameters:
+ *   vtable - Pointer to framebuffer's virtual table.
+ *
+ ****************************************************************************/
+
+void fb_pollnotify(FAR struct fb_vtable_s *vtable)
+{
+  FAR struct fb_chardev_s *fb;
+
+  DEBUGASSERT(vtable != NULL);
+
+  fb = vtable->priv;
+
+  /* If there is no listener, it is not processed. */
+
+  if (fb->fds == NULL)
```

Comment:
don't need since poll_notify check it internally

Reply:
done